Extension/src/LanguageServer/referencesProvider.ts

```diff
@@ -47,119 +47,118 @@ function getReferenceItemIconPath(type: ReferenceType, isCanceled: boolean): { l
     return (isCanceled && type === ReferenceType.ConfirmationInProgress) ? getReferenceCanceledIconPath() : getReferenceTypeIconPath(type);
 }
 
-type TreeObject = FileItem | ReferenceItem | ReferenceTypeItem;
-
-export class ReferenceDataProvider implements vscode.TreeDataProvider<TreeObject> {
-    private references: Model;
-    private referencesCanceled: boolean = false;
-    private readonly _onDidChangeTreeData = new vscode.EventEmitter<TreeObject>();
+export class ReferenceDataProvider implements vscode.TreeDataProvider<TreeNode> {
+    private referencesModel: ReferencesModel;
+    private readonly _onDidChangeTreeData = new vscode.EventEmitter<TreeNode>();
     readonly onDidChangeTreeData = this._onDidChangeTreeData.event;
-    private groupByFile: boolean = true;
 
-    constructor() {
-        vscode.commands.executeCommand('setContext', 'refView.isGroupedByFile', true);
+    constructor(readonly isCandidates: boolean) {

```

Comment:
>isCandidates [](start = 25, length = 12)

nit: how about "isRenameCandidates" so it's more descriptive?

Reply:
also, should "isCandidates" be owned by the TreeNode object?

---
In reply to: [332144586](https://github.com/Microsoft/vscode-cpptools/pull/4378#discussion_r332144586) [](ancestors = 332144586)